presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -89,23 +90,81 @@ public static Slice replace(@SqlType(StandardTypes.VARCHAR) Slice str, @SqlType(
     @Description("greedily replaces occurrences of a pattern with a string")
     @ScalarFunction
     @SqlType(StandardTypes.VARCHAR)
-    public static Slice replace(@SqlType(StandardTypes.VARCHAR) Slice str, @SqlType(StandardTypes.VARCHAR) Slice search, @SqlType(StandardTypes.VARCHAR) Slice replace)
+    public static Slice replace(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType(StandardTypes.VARCHAR) Slice search, @SqlType(StandardTypes.VARCHAR) Slice replace)
     {
-        String replaced = str.toString(UTF_8).replace(
-                search.toString(UTF_8),
-                replace.toString(UTF_8));
-        return Slices.copiedBuffer(replaced, UTF_8);
+        //
+        // Empty search returns original string
+        if (UnicodeUtil.isEmpty(search)) {
+            return string;
```

Comment:
Let's keep this backward-compatible. `replace('abc', '', 'xx')` produces `xxaxxbxxcxx`. This also seems more intuitive to me.